src/Stratis.Bitcoin/P2P/PeerConnectorContext.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.P2P
+{
+    /// <summary>Context class for <see cref="PeerConnector"/>.</summary>
+    public sealed class PeerConnectorContext
```

Comment:
I have an issue with putting a Context here, the types in this context are all singletons available in the DI.
A Context I normally use as a type that hold state for a duration of an operation.

I would propose to just use our normal pattern of passing Interfaces in to the PeerConenctor, and for testing just create a PeerTestFactory that creates a PeerConnector with all the interfaces mocked away.

Reply:
OK I'll revert this then 👍